pmm-app/src/pmm-settings/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,126 @@
+};
+
+export const SignUp: FC<SignUpProps> = ({ userEmail }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+  const [loggedInEmail, setLoggedInEmail] = useState(userEmail);
+
+  const CheckboxLabel = () => (
+    <span data-qa="sign-up-agreement-checkbox-label" className={styles.checkboxLabel}>
+      {Messages.agreementFirstPart}
+      {' '}
+      <LinkButton className={styles.link} variant="link" href={TERMS_OF_SERVICE_URL}>{Messages.termsOfService}</LinkButton>
+      ,
+      {' '}
+      <LinkButton className={styles.link} variant="link" href={PRIVACY_POLICY_URL}>{Messages.privacyPolicy}</LinkButton>
+      ,
+      {' '}
+      {Messages.agreementSecondPart}
+      {' '}
+      <LinkButton className={styles.link} variant="link" href={NOTIFICATION_SETTINGS_URL}>{Messages.notificationSettings}</LinkButton>
+    </span>
+  );
+
+  const handleSignUpFormSubmit = async (signUpData: SignUpData) => {
+    try {
+      await signUp(signUpData);
+      setLoggedInEmail(signUpData.email);
+      showSuccessNotification({ message: Messages.signUpSucceeded });
+    } catch (e) {
+      console.error(e);
+      showErrorNotification({ message: Messages.errors.signUpFailed });
+    }
+  };
+
+  const InnerForm: FC<FormRenderProps<SignUpData>> = ({
+    pristine, submitting, valid, handleSubmit
+  }) => (
+    <form data-qa="sign-up-form" className={styles.form} onSubmit={handleSubmit}>
+      <legend className={styles.legend}>
+        {Messages.signUp}
+      </legend>
+      <FinalField
+        data-qa="sign-up-email-input"
+        name="email"
+        label={Messages.emailLabel}
+        component={InputFieldAdapter}
+        validate={validators.compose(validators.required, validators.validateEmail)}
+      />
+      <FinalField
+        data-qa="sign-up-password-input"
+        name="newPassword"
+        label={Messages.passwordLabel}
+        type="password"
+        component={InputFieldAdapter}
+        validate={validators.required}
+        autoComplete="on"
+      />
+      <FinalField
+        data-qa="sign-up-agreement-checkbox"
+        className={styles.checkboxWrapper}
+        label={<CheckboxLabel />}
+        name="agreement"
+        component={CheckboxFieldAdapter}
+        validate={validators.required}
+        type="checkbox"
+      />
+      <ButtonWithSpinner
+        data-qa="sign-up-submit-button"
+        className={styles.submitButton}
+        type="submit"
+        disabled={!valid || submitting || pristine}
+        isLoading={submitting}
+      >
+        {Messages.signUp}
+      </ButtonWithSpinner>
+      <Button
+        data-qa="sign-up-to-sign-in-button"
+        className={styles.signInButton}
+        type="button"
+        variant="destructive"
+      >
+        {Messages.signIn}
+      </Button>
+    </form>
+  );
+
+  return (
+    // TODO (nicolalamacchia): improve this once the sign in component is ready
+    <>
+      {typeof loggedInEmail !== 'undefined'
```

Comment:
I guess a much better approach would be to just check the value.
```suggestion
      {loggedInEmail != null
```

Reply:
Done